browser/brave_stats/brave_stats_updater_browsertest.cc

```diff
@@ -76,187 +85,117 @@ class BraveStatsUpdaterBrowserTest : public InProcessBrowserTest {
-  stats_updater.SetStatsUpdatedCallback(
-      base::BindRepeating(&BraveStatsUpdaterBrowserTest::OnStandardStatsUpdated,
-                          base::Unretained(this)));
-  stats_updater.Start();
   WaitForStandardStatsUpdatedCallback();
-  // Normal usage ping is _always_ sent before the threshold ping.
-  EXPECT_TRUE(StandardStatsEndpointWasCalled());
-  stats_updater.Stop();
-
-  // Stop the referrals service
-  referrals_service->Stop();
 
   // First check preference should now be true
-  EXPECT_TRUE(GetLocalState()->GetBoolean(kFirstCheckMade));
+  EXPECT_TRUE(g_browser_process->local_state()->GetBoolean(kFirstCheckMade));
 }
 
 // Run the stats updater and verify the threshold endpoint is reached
 IN_PROC_BROWSER_TEST_F(BraveStatsUpdaterBrowserTest,
                        StatsUpdaterThresholdSetsFirstCheckPreference) {
-  // Ensure that first check preference is false
-  ASSERT_FALSE(GetLocalState()->GetBoolean(kFirstCheckMade));
-
-  // Start the referrals service, since the stats updater's startup
-  // ping only occurs after the referrals service checks for the promo
-  // code file
-  auto referrals_service =
-      brave::BraveReferralsServiceFactory::GetInstance()->GetForPrefs(
-          GetLocalState());
-  referrals_service->SetReferralInitializedCallbackForTest(
-      base::BindRepeating(&BraveStatsUpdaterBrowserTest::OnReferralInitialized,
-                          base::Unretained(this)));
-  referrals_service->Start();
-  WaitForReferralInitializeCallback();
+  EXPECT_TRUE(
+      g_brave_browser_process->brave_stats_updater()->MaybeDoThresholdPing(3));
 
-  // Start the stats updater, wait for it to perform its startup ping,
-  // and then shut it down
-  brave_stats::BraveStatsUpdater stats_updater(GetLocalState());
-  stats_updater.SetStatsThresholdCallback(base::BindRepeating(
-      &BraveStatsUpdaterBrowserTest::OnThresholdStatsUpdated,
-      base::Unretained(this)));
-  stats_updater.Start();
-  EXPECT_TRUE(stats_updater.MaybeDoThresholdPing(3));
+  WaitForReferralInitializeCallback();
   WaitForThresholdStatsUpdatedCallback();
-  stats_updater.Stop();
-
-  // Stop the referrals service
-  referrals_service->Stop();
 
   // First check and Threshold check should be set.
-  EXPECT_TRUE(GetLocalState()->GetBoolean(kFirstCheckMade));
-  EXPECT_TRUE(GetLocalState()->GetBoolean(kThresholdCheckMade));
+  EXPECT_TRUE(g_browser_process->local_state()->GetBoolean(kFirstCheckMade));
+  EXPECT_TRUE(
+      g_browser_process->local_state()->GetBoolean(kThresholdCheckMade));
 }
 
 // Run the stats updater with no active referral and verify that the
 // update url specifies the default referral code
 IN_PROC_BROWSER_TEST_F(BraveStatsUpdaterBrowserTest,
                        StatsUpdaterStartupPingWithDefaultReferralCode) {
-  // Ensure that checked for promo code file preference is false
-  ASSERT_FALSE(GetLocalState()->GetBoolean(kReferralInitialization));
```

Comment:
And again :) (re: `EXPECT_FALSE(g_browser_process->local_state()->GetBoolean(kFirstCheckMade));`)

Reply:
I obviously meant `EXPECT_FALSE(g_browser_process->local_state()->GetBoolean(kReferralInitialization))` in this case though!